File: LHDS.ConfigImportExportTool/Models/Foundations/SpecificationObjects/SpecificationObject.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;
using System.Collections.Generic;
using LHDS.ConfigImportExportTool.Models.Foundations.DatasetSpecifications;
using LHDS.ConfigImportExportTool.Models.Foundations.ObjectColumns;

namespace LHDS.ConfigImportExportTool.Models.Foundations.SpecificationObjects
{
    public class SpecificationObject
    {
        public Guid Id { get; set; }
        public Guid DataSetSpecificationId { get; set; }
        public string SupplierObjectName { get; set; } = string.Empty;
        public string OurObjectName { get; set; } = string.Empty;
        public string ObjectDescription { get; set; } = string.Empty;
        public string InterchangeProtocol { get; set; } = string.Empty;
        public bool IsPushedToUs { get; set; }
        public bool IsPulledByUs { get; set; }
        public string DeletionHandling { get; set; } = string.Empty;
        public bool IsSubmissionHeaderObject { get; set; }
        public bool IsTransactionLog { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public string UpdatedBy { get; set; } = string.Empty;
        public DateTimeOffset UpdatedDate { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public DataSetSpecification DataSetSpecification { get; set; } = null!;
        public List<ObjectColumn> ObjectColumns { get; set; } = new List<ObjectColumn>();
    }
}

File: LHDS.AdminPortal.Configurations/lhds.adminportal.configurations.client/src/models/specificationObjects/specificationObject.ts
import { DataSetSpecification } from '../dataSetSpecifications/dataSetSpecification';

export class SpecificationObject {
    public id: string;
    public dataSetSpecificationId: string;
    public supplierObjectName: string;
    public ourObjectName: string;
    public objectDescription: string;
    public interchangeProtocol: string;
    public isPushedToUs: boolean;
    public isPulledByUs: boolean;
    public deletionHandling: string;
    public isSubmissionHeaderObject: boolean;
    public isTransactionLog: boolean
    public createdBy?: string;
    public createdDate?: Date;
    public updatedBy?: string;
    public updatedDate?: Date;
    public dataSetSpecification?: DataSetSpecification;

    constructor(dataSetObjects: SpecificationObject) {
        this.id = dataSetObjects.id ? dataSetObjects.id : "";
        this.dataSetSpecificationId = dataSetObjects.dataSetSpecificationId;
        this.supplierObjectName = dataSetObjects.supplierObjectName;
        this.ourObjectName = dataSetObjects.ourObjectName;
        this.objectDescription = dataSetObjects.objectDescription;
        this.interchangeProtocol = dataSetObjects.interchangeProtocol;
        this.isPushedToUs = dataSetObjects.isPushedToUs;
        this.isPulledByUs = dataSetObjects.isPulledByUs;
        this.deletionHandling = dataSetObjects.deletionHandling;
        this.isSubmissionHeaderObject = dataSetObjects.isSubmissionHeaderObject;
        this.isTransactionLog = dataSetObjects.isTransactionLog;
        this.createdDate = dataSetObjects.createdDate;
        this.createdBy = dataSetObjects.createdBy;
        this.createdDate = new Date(dataSetObjects.createdDate);
        this.updatedBy = dataSetObjects.updatedBy;
        this.updatedDate = new Date(dataSetObjects.updatedDate);

        if (dataSetObjects.dataSetSpecification !== undefined && dataSetObjects.dataSetSpecification !== null) {
            this.dataSetSpecification = new DataSetSpecification(dataSetObjects.dataSetSpecification);
        }
    }
}